File: Programming EF Code First/chapter5/Chapter 5 VB/Fluent/DataAccess/BreakAwayContext.vb
Imports System.Data.Entity
Imports Model

Namespace DataAccess
  Public Class BreakAwayContext
	  Inherits DbContext
	Public Property Destinations() As DbSet(Of Destination)
	Public Property Lodgings() As DbSet(Of Lodging)
	Public Property Trips() As DbSet(Of Trip)
	Public Property People() As DbSet(Of Person)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
	  ' Entity Type Configuration
	  modelBuilder.Configurations.Add(New DestinationConfiguration())
	  modelBuilder.Configurations.Add(New LodgingConfiguration())
	  modelBuilder.Configurations.Add(New TripConfiguration())
	  modelBuilder.Configurations.Add(New PersonConfiguration())
	  modelBuilder.Configurations.Add(New InternetSpecialConfiguration())
	  modelBuilder.Configurations.Add(New ActivityConfiguration())
	  modelBuilder.Configurations.Add(New PersonPhotoConfiguration())
	  modelBuilder.Configurations.Add(New ReservationConfiguration())

	  ' Complex Type Configuration
	  modelBuilder.Configurations.Add(New AddressConfiguration())
	  modelBuilder.ComplexType(Of PersonalInfo)()
	End Sub
  End Class
End Namespace

File: Programming EF Code First/chapter2/Chapter2/BAGA_FluentAPI/DataAccess/BreakAwayContext.cs
using System.Data.Entity;
using Model;

namespace DataAccess
{
  public class BreakAwayContext : DbContext
  {
    public DbSet<Destination> Destinations { get; set; }
    public DbSet<Lodging> Lodgings { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder)
    {
      modelBuilder.Configurations.Add(new DestinationConfiguration());
      modelBuilder.Configurations.Add(new LodgingConfiguration());
    }
  }
}